File: frontend/src/types/health.ts
export type ComponentStatus = "healthy" | "unavailable";

export interface ComponentHealth {
  status: ComponentStatus;
  latency_ms: number | null;
}

export interface HealthResponse {
  status: "healthy" | "degraded";
  service: string;
  version: string;
  build_sha: string | null;
  build_time: string | null;
  environment: string;
  checks: Record<string, ComponentHealth>;
}

File: backend/app/schemas/health.py
from typing import Literal

from pydantic import BaseModel, Field


class ComponentHealth(BaseModel):
    status: Literal["healthy", "unavailable"]
    latency_ms: float | None = Field(default=None, ge=0)


class HealthResponse(BaseModel):
    status: Literal["healthy", "degraded"]
    service: str
    version: str
    build_sha: str | None = None
    build_time: str | None = None
    environment: str
    checks: dict[str, ComponentHealth]
